.scalafmt.conf

```diff
@@ -1,4 +1,4 @@
-version = "2.5.2"
+version = "2.5.3"
```

Comment:
If I get this right, ths configuration key is used by `sbt-scalamt` to dynamically load `scalafix-core`. However, the CI & [documented](https://github.com/scalacenter/sbt-scalafix/blob/master/CONTRIBUTING.md#formatting) way to run scalafmt in this repo is through the CLI `bin/scalafmt`, which from my understanding does not read this. Therefore I am wondering if this is not dead conf that we could remove?

Alternatively, maybe we could use `sbt-scalafmt` to get rid of the opaque binary in `bin/` and benefit from scala-steward updates? WDYT @olafurpg ?

Reply:
The cli binary, sbt-scalafmt, IntelliJ and Metals all use the same logic to dynamically load the correct version of Scalafmt from this config via `org.scalameta:scalafmt-dynamic`. I personally like being able to run formatting from the terminal instead of the sbt console. It's pretty clear I'm in a minority since most people seem to prefer sbt-scalafmt. 

I'm OK with adding sbt-scalafmt if you prefer to format via the sbt console, but I would prefer to also keep the binary.

Reply:
Thanks for the explanation! I think I got confused by the doc and the fact that the version in `.scalafmt.conf` was optional. I opened https://github.com/scalameta/scalafmt/pull/2040 to try to clarify for others.

As for CLI vs the sbt plugin,I don't have a strong preference, I just wanted to make sure this bump actually had some effect (and wanted to understand when/if bumping the binary was necessary), so let's keep it as-is :+1: